src/cryptography/hazmat/primitives/asymmetric/rsa.py

```diff
@@ -42,13 +42,30 @@ def public_key(self):
 
 
 @six.add_metaclass(abc.ABCMeta)
-class RSAPrivateKeyWithNumbers(RSAPrivateKey):
+class RSAPrivateKeyWithSerialization(RSAPrivateKey):
```

Comment:
Leave the subclassing orthagonal IMO; no need to change more things than
our necessary with this patch.

On Sat, Feb 28, 2015 at 4:45 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/primitives/asymmetric/rsa.py
> <https://github.com/pyca/cryptography/pull/1503#discussion_r25562882>:
>
> > @@ -42,13 +42,30 @@ def public_key(self):
> >
> >
> >  @six.add_metaclass(abc.ABCMeta)
> > -class RSAPrivateKeyWithNumbers(RSAPrivateKey):
> > +class RSAPrivateKeyWithSerialization(RSAPrivateKey):
>
> Not really no. RSAPrivateKeyWithNumbers is already a subclass so I'd
> oppose changing that, but we could certainly choose to make its successor (
> WithSerialization) not a subclass and simply register both interfaces.
> The numbers classes are (I believe) the only place in cryptography that we
> utilize inheritance.
>
> @alex <https://github.com/alex>, @public <https://github.com/public> got
> opinions here? I'm happy to go either way.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1503/files#r25562882>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084